cdap-gateway/src/main/java/co/cask/cdap/gateway/router/NettyRouter.java

```diff
@@ -157,15 +156,16 @@ public NettyRouter(CConfiguration cConf, SConfiguration sConf, @Named(Constants.
   protected void startUp() throws Exception {
     ChannelUpstreamHandler connectionTracker = new SimpleChannelUpstreamHandler() {
       @Override
-      public void handleUpstream(ChannelHandlerContext ctx, ChannelEvent e)
+      public void channelOpen(ChannelHandlerContext ctx, ChannelStateEvent e)
         throws Exception {
         channelGroup.add(e.getChannel());
-        super.handleUpstream(ctx, e);
+        super.channelOpen(ctx, e);
```

Comment:
Channel is first open, then bound to a local address and then connected to remote peer. If we close all open channels then it should take care of all connected or bound channels.